pnc-rest/src/main/java/org/jboss/pnc/rest/endpoint/BuildConfigurationEndpoint.java

```diff
@@ -56,6 +56,9 @@ public BuildConfigurationEndpoint(BuildConfigurationProvider buildConfigurationP
     @ApiOperation(value = "Creates a new Build Configuration")
     @POST
     public Response createNew(@NotNull @Valid BuildConfigurationRest buildConfigurationRest, @Context UriInfo uriInfo) {
+        if (buildConfigurationRest.getId() != null) {
```

Comment:
Consider using Guava's Predications in buildConfigurationProvider#store. This way the provider could be reused and IllegalArgumentException will be automatically translated into HTTP 400